dashboard/test/controllers/api/v1/sections_controller_test.rb

```diff
@@ -584,4 +584,42 @@ def returned_json
   def returned_section
     Section.find returned_json['id']
   end
+
+  test "update_sharing_disabled updates sharing_disabled" do
+    sign_in @teacher
+    section = create(:section, user: @teacher, script_id: Script.flappy_script.id)
+    post :update_sharing_disabled, params: {
+      id: section.id,
+      sharing_disabled: true
+    }
+    assert_response :success
+    section.reload
+    assert_equal(true, section.sharing_disabled)
+
+    post :update_sharing_disabled, params: {
+      id: section.id,
+      sharing_disabled: false
+    }
+    assert_response :success
+    section.reload
+    assert_equal(false, section.sharing_disabled)
+  end
+
+  test "update_sharing_disabled: cannot update section you dont own" do
+    other_teacher = create(:teacher)
+    sign_in other_teacher
+    post :update_sharing_disabled, params: {
+      id: @section.id,
+      sharing_disabled: true,
+    }
+    assert_response :forbidden
+  end
```

Comment:
FYI we have a [helper](https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/test/test_helper.rb#L325) for these boiler plate tests.

If you want this can be
```ruby
test_user_gets_response_for :update_sharing_disabled, user: :teacher, params: -> {{id: @section.id, sharing_disabled: true}}, response: :forbidden
```